### PR TITLE
Use newer ReJSON API

### PR DESCRIPTION
**Pull request by ashtul**
This might not pass tests before RedisJSON/RedisJSON#417 is merged

### CODE REVIEW

File: tests/pytests/test_json.py [hunk 1]
```diff
@@ -201,6 +201,11 @@ def testSet(env):
     # Can also do multiple changes/side-effects, such as converting an object to a scalar
     env.execute_command('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA', '$.t', 'TEXT')
```
Comment (oshadmi) [inline tests/pytests/test_json.py:202]: Need to add some wildcard/union/slice/recursive-desent to a path in order to get multiple values
(for example, [testMultiPathResults](https://github.com/RedisJSON/RedisJSON/blob/8813a86e46765f3c411e5cc59bda704e9379fa60/tests/pytest/test.py#L859))

**Review by oshadmi — COMMENTED** [review]
😎 Few comments

File: src/document_basic.c [hunk 2]
```diff
@@ -167,20 +167,19 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
...
-    // TODO: check option to move to getStringFromKey
-    json = japi->get(jsonKey, field->path, &jsonType);
-    if (!json) {
+    jsonIter = japi->get(jsonKey, field->path);
+    if (!jsonIter) {
         continue;
     }
 
-    if (FieldSpec_CheckJsonType(field->types, jsonType) != REDISMODULE_OK) {
+    RedisJSON jsonValue = japi->next(jsonIter);
```
[the remark this answers is not in the document]
Comment (ashtul) [reply src/document_basic.c:180]: @oshadmi This is not the multivalue PR but an interim PR to align RediSearch with RedisJSON

File: src/rlookup.c [hunk 3]
```diff
@@ -432,18 +433,18 @@ static int getKeyCommonJSON(const RLookupKey *kk, RLookupRow *dst, RLookupLoadOp
     path = fs ? fs->path : kk->name;
   }
 
-  JSONType jsonType;
-  RedisJSON json = japi->get(*keyobj, path, &jsonType);
-
-  if (!json) {
+  JSONResultsIterator jsonIter = japi->get(*keyobj, path);
```
[the remark this answers is not in the document]
Comment (ashtul) [reply src/rlookup.c:436]: Same

File: src/document_basic.c [hunk 4]
```diff
@@ -154,9 +154,9 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
   SchemaRule *rule = spec->rule;
   RedisModuleCtx *ctx = sctx->redisCtx;
   size_t nitems = sctx->spec->numFields;
-  RedisJSON json = NULL;
+  JSONResultsIterator jsonIter = NULL;
 
-  RedisJSONKey jsonKey = japi->openKey(ctx, doc->docKey);
+  RedisJSON jsonKey = japi->openKey(ctx, doc->docKey);
```
Comment (oshadmi) [inline src/document_basic.c:159]:
```suggestion
  RedisJSON jsonRoot = japi->openKey(ctx, doc->docKey);
```

**Review by oshadmi — APPROVED** [review]
👍🏼 LGTM

**Pull request merged**